src/Raven.Server/Documents/Sharding/Subscriptions/ShardedSubscriptionWorker.cs

```diff
@@ -61,7 +64,28 @@ protected override async Task SendAckAsync(ShardedSubscriptionBatch batch, Strea
             }
         }
 
-        protected override RequestExecutor GetRequestExecutor() => _shardRequestExecutor;
+        protected override Task TrySetRedirectNode(RequestExecutor reqEx, IReadOnlyList<ServerNode> curTopology)
+        {
+            if (_databaseContext.ShardsTopology.TryGetValue(_shardNumber, out var topology))
+            {
+                var node = topology.WhoseTaskIsIt(RachisState.Follower, _state.SubscriptionState, null);
+                if (node == null)
+                    return Task.CompletedTask;
+
+                _redirectNode = _shardRequestExecutor.TopologyNodes.FirstOrDefault(x => x.ClusterTag == node);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        protected override RequestExecutor GetRequestExecutor()
+        {
+            if (_shardRequestExecutor == null || _shardRequestExecutor.Disposed)
+            {
+                _shardRequestExecutor = _databaseContext.ShardExecutor.GetRequestExecutorAt(_shardNumber);
```

Comment:
shouldn't you set it outside?

Reply:
for `ShardedSubscriptionWorker` we always return `_shardRequestExecutor` this code is just to refresh it in case its disposed